python/ray/tune/automl/search_policy.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python
+# Author: Rujie Jiang rujie.jrj@antfin.com
+# Date: Tue May 15 11:46:20 2018
+
+import time
+import copy
+import logging
+
+from ray.tune.trial import Trial
+from ray.tune.suggest import SearchAlgorithm
+from ray.tune.suggest.variant_generator import generate_variants
+from ray.tune.config_parser import make_parser, create_trial_from_spec
+
+
+def merge_config(path, value, config):
+    """Merge kv pair into config by expanding path separated by dot."""
+    keys = path.split('.')
+    for name in keys[:-1]:
+        if config.get(name) is None:
+            config[name] = {}
+        config = config[name]
+    # Set value corresponding to the last path name
+    config[keys[-1]] = value
+
+
+class AutoMLSearcher(SearchAlgorithm):
```

Comment:
In which way exactly is this searcher different from any other general searcher? It seems it is mostly a searcher to search `SearchSpace`? All AutoML logic is embedded in `SearchSpace`, no? This searcher is then just calling into it and preparing trials?

Reply:
This PR contains just one tip (the genetic search algorithm) of our project, since we think it would be general enough to enhance the ray.tune.

As for the AutoMLSearcher, it's main job is to search within the `SearhSpace`. However the AutoML logic (how `SearchSpace` is generated) resides the AutoML framework, which we haven't contribute yet.